avahi-client/client-test.c

```diff
@@ -272,35 +279,36 @@ int main (AVAHI_GCC_UNUSED int argc, AVAHI_GCC_UNUSED char *argv[]) {
     assert(r >= 0);
     assert(avahi_string_list_serialize(txt, NULL, 0) == sizeof(rdata));
     error = avahi_entry_group_add_service_strlst(group, AVAHI_IF_UNSPEC, AVAHI_PROTO_UNSPEC, 0, "TestX", "_qotd._tcp", NULL, NULL, 123, txt);
-    assert(error == AVAHI_ERR_INVALID_RECORD);
+    //assert(error == AVAHI_ERR_INVALID_RECORD);
```

Comment:
Why has this been commented out?

Reply:
I'm guessing it's because it was initially used to reproduce the CNAME bug locally by editing the line above. When the avahi crash was gone and the other tests were added it was probably left here accidentally.

(This PR isn't mergeable currently so there are various things that probably came from debugging sessions and need fixing)